docs/src/users_guide/steadystate.md

```diff
@@ -0,0 +1,133 @@
+    It is necessary to provide the initial state `ψ0` for ODE solving method, namely
+    `steadystate(H, ψ0, tspan, c_ops, solver = SteadyStateODESolver())`, where `tspan::Real` represents the final time step, defaults to `Inf` (infinity).
+
+Although it is not obvious, the [`SteadyStateDirectSolver()`](@ref SteadyStateDirectSolver) and [`SteadyStateEigenSolver()`](@ref SteadyStateEigenSolver) methods all use an LU decomposition internally and thus can have a large memory overhead. In contrast, for [`SteadyStateLinearSolver()`](@ref SteadyStateLinearSolver), iterative algorithms provided by [`LinearSolve.jl`](https://docs.sciml.ai/LinearSolve/stable/solvers/solvers/), such as `KrylovJL_GMRES()` and `KrylovJL_BICGSTAB()`, do not factor the matrix and thus take less memory than the LU methods and allow, in principle, for extremely large system sizes. The downside is that these methods can take much longer than the direct method as the condition number of the Liouvillian matrix is large, indicating that these iterative methods require a large number of iterations for convergence. 
+
+To overcome this, one can provide preconditioner that solves for an approximate inverse for the (modified) Liouvillian, thus better conditioning the problem, leading to faster convergence. The left and right preconditioner can be specified as the keyword argument `Pl` and `Pr`, respectively:
+```julia
+solver = SteadyStateLinearSolver(alg = KrylovJL_GMRES(), Pl = Pl, Pr = Pr)
+```
+The use of a preconditioner can actually make these iterative methods faster than the other solution methods. The problem with precondioning is that it is only well defined for Hermitian matrices. Since the Liouvillian is non-Hermitian, the ability to find a good preconditioner is not guaranteed. Moreover, if a preconditioner is found, it is not guaranteed to have a good condition number. 
+
+Furthermore, `QuantiumToolbox` can take advantage of the Intel (Pardiso) LU solver in the Intel Math Kernel library (Intel-MKL) by using [`LinearSolve.jl`](https://docs.sciml.ai/LinearSolve/stable/) and either [`Pardiso.jl`](https://github.com/JuliaSparse/Pardiso.jl) or [`MKL_jll.jl`](https://github.com/JuliaBinaryWrappers/MKL_jll.jl):
+
+```julia
+using QuantumToolbox
+using LinearSolve # must be loaded
+
+using Pardiso
+solver = SteadyStateLinearSolver(alg = MKLPardisoFactorize())
+
+using MKL_jll
+solver = SteadyStateLinearSolver(alg = MKLLUFactorization())
+```
+
+See [`LinearSolve.jl` Solvers](https://docs.sciml.ai/LinearSolve/stable/solvers/solvers/) for more details.
+
+## Example: Harmonic Oscillator in Thermal Bath
+
+```@example steady_state_example
+using QuantumToolbox
+using CairoMakie
+CairoMakie.enable_only_mime!(MIME"image/svg+xml"())
+
+
+# Define parameters
+N = 20  # number of basis states to consider
+a = destroy(N)
+H = a' * a
+ψ0 = basis(N, 10)  # initial state
+κ = 0.1  # coupling to oscillator
+
+# collapse operators
+c_op_list = QuantumObject[]
```

Comment:
Predefining an empty vector and then pushing into it is fine. However, if doing so, we should define the entire type to make it type-stable. In this case, the compile would only know that `c_op_list` is a `Vector{QuantumObject}` and not something more specific like `Vector{QuantumObject{SparseMatrixCSC{ComplexF64, Int64}, Operator, 1}`, so it would lead to type instabilities. 

I know that this is just an example, but I would always show type-stable examples. In this way, a new user that simply performs copy and paste, will have already a type stable code.

In this case we don't need to push. `\gamma` is already different from zero. Such that we can make the code evens simpler.

Reply:
okay !